astropy/modeling/functional_models.py

```diff
@@ -1054,49 +1062,84 @@ class Voigt1D(Fittable1DModel):
+                fwhm_G=fwhm_G.default, method='hum2zpf16c', n_models=None, **kwargs):
+        if str(method).lower() in ('wofz', 'faddeeva', 'scipy'):
+            try:
+                from scipy.special import wofz
+            except (ValueError, ImportError) as err:
+                raise ImportError(f'Voigt1D method {method} requires scipy: {err}.') from err
+            cls._wofz = wofz
+        elif str(method).lower() in ('humlicek2', 'hum2zpf16c'):
+            cls._wofz = cls.hum2zpf16c
+        else:
+            raise ValueError(f'Not a valid method for Voigt1D Faddeeva function: {method}.')
 
-    @classmethod
-    def evaluate(cls, x, x_0, amplitude_L, fwhm_L, fwhm_G):
+        return super().__new__(cls, **kwargs)
+
+    def __init__(self, x_0=x_0.default, amplitude_L=amplitude_L.default, fwhm_L=fwhm_L.default,
+                 fwhm_G=fwhm_G.default, method='hum2zpf16c', n_models=None, **kwargs):
+        super().__init__(x_0=x_0, amplitude_L=amplitude_L, fwhm_L=fwhm_L, fwhm_G=fwhm_G, **kwargs)
 
-        A, B, C, D = cls._abcd
-        sqrt_ln2 = np.sqrt(np.log(2))
-        X = (x - x_0) * 2 * sqrt_ln2 / fwhm_G
-        X = np.atleast_1d(X)[..., np.newaxis]
-        Y = fwhm_L * sqrt_ln2 / fwhm_G
-        Y = np.atleast_1d(Y)[..., np.newaxis]
+    @classmethod
+    def faddeeva(cls, z, method=None):
+        """Call complex error (Faddeeva) function w(z) implemented by algorithm 'method';
+        cache results for consecutive calls from `evaluate`, `fit_deriv`."""
+
+        if method is not None:
+            if str(method).lower() in ('wofz', 'faddeeva', 'scipy'):
+                try:
+                    from scipy.special import wofz
+                except (ValueError, ImportError) as err:
+                    raise ImportError(f'Voigt1D method {method} requires scipy: {err}.') from err
+                cls._wofz = wofz
+            elif str(method).lower() in ('humlicek2', 'hum2zpf16c'):
+                cls._wofz = cls.hum2zpf16c
+            else:
+                raise ValueError(f'Not a valid method for Voigt1D Faddeeva function: {method}.')
+
+        if (cls._wofz == cls._last_wofz and z.shape == cls._last_z.shape and
+            np.allclose(z, cls._last_z, rtol=1.e-14, atol=1.e-15)):
+            return cls._last_w
+
+        cls._last_w = cls._wofz(z)
+        cls._last_z = z
+        cls._last_wofz = cls._wofz
+        return cls._last_w
 
-        V = np.sum((C * (Y - A) + D * (X - B))/((Y - A) ** 2 + (X - B) ** 2), axis=-1)
+    @classmethod
+    def evaluate(cls, x, x_0, amplitude_L, fwhm_L, fwhm_G, method=None):
+        """One dimensional Voigt function scaled to Lorentz peak amplitude."""
 
-        return (fwhm_L * amplitude_L * np.sqrt(np.pi) * sqrt_ln2 / fwhm_G) * V
+        z = np.atleast_1d(2 * (x - x_0) + 1j * fwhm_L) * cls.sqrt_ln2 / fwhm_G
+        # The normalised Voigt profile is w.real * cls.sqrt_ln2 / (cls.sqrt_pi * fwhm_G) * 2 ;
```

Comment:
super minor: astropy docs are in US English, so
```suggestion
        # The normalized Voigt profile is w.real * cls.sqrt_ln2 / (cls.sqrt_pi * fwhm_G) * 2 ;
```